app/api/submit-message/route.ts

```diff
@@ -0,0 +1,64 @@
+import OpenAI from "openai";
+
+export const POST = async (req: Request): Promise<Response> => {
+  if (req.method !== "POST") {
+    const responseData = JSON.stringify({ error: "Method not allowed" });
+    return new Response(responseData, {
+      status: 405,
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+  }
+
+  const body = await req.json();
+
+  try {
+    const { assistantId, threadId, message } = body;
+
+    const openai = new OpenAI();
+
+    if (!process.env.OPENAI_API_KEY) {
+      throw new Error("Missing env var from OpenAI");
+    }
+
+    await openai.beta.threads.messages.create(threadId, message);
+
+    const run = await openai.beta.threads.runs.create(threadId, {
+      assistant_id: assistantId,
+    });
+
+    let runStatus = null;
+
+    do {
+      runStatus = await openai.beta.threads.runs.retrieve(threadId, run.id);
+      // Sleep for a short time before checking again
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+    } while (runStatus.status !== "completed");
+
+    const messages = await openai.beta.threads.messages.list(threadId);
+
+    const formattedMessages = messages.data.map((msg) => ({
+      role: msg.role,
+      content: msg.content[0].text.value,
```

Comment:
we made need to address this as there is a false positive typescript error
see from issue notes below: 

_To be direct, this isn't an incomplete type – this is TypeScript being pedantic about the fact that an element of the content array could either be {type: 'text', text: string} or {type: 'image_url', image_url: string} and if it's the latter, the text property won't exist.
You need to first check whether type === 'text' before accessing .text.
In other words, while this can be toilsome, it is working as intended._

`  const messages = data.map((message) =>
    message?.content.map((content) => (content.type === 'text' ? content.text : '')).join(''),
  );`